File: src/utils/window.ts
import { appWindow, LogicalSize, LogicalPosition } from '@tauri-apps/api/window'

export const SLIDE_MODE_WINDOW_SIZE = {
    width: 800,
    height: 1200
}

export const setOnTop = async () => {
    await appWindow.setAlwaysOnTop(true)
}

export const setNotOnTop = async () => {
    await appWindow.setAlwaysOnTop(false)
}

export const moveWindow = async (x: number, y: number) => {
    await appWindow.setPosition(new LogicalPosition(x, y))
}

export const resizeWindow = async (width: number, height: number) => {
    await appWindow.setSize(new LogicalSize(width, height))
}

export const hideWindow = () => {
    appWindow.hide()
}

File: src-tauri/src/window.rs
use tauri::{AppHandle, Event, Manager, PhysicalPosition, PhysicalSize};
use crate::entities::MoveToAndSetOnTopPayload;
use crate::event::EVENT_SLIDE_FOCUS_INPUT;

pub fn move_to_and_set_on_top(app_handle: &AppHandle, event: Event) {
    let payload_str = event.payload().unwrap();
    let payload = serde_json::from_str::<MoveToAndSetOnTopPayload>(payload_str);
    
    if let Ok(payload) = payload {
        // resize window
        let (width, height) = (payload.width, payload.height);
        let (x, y) = (payload.x, payload.y);

        let window = app_handle.get_window("main").unwrap();

        window.set_size(PhysicalSize{
            width: width,
            height: height
        }).unwrap();

        // Move the window
        window.set_position(PhysicalPosition {
            x: payload.x,
            y: payload.y,
        }).unwrap();

        // Set the window always on top
        window.set_always_on_top(true).unwrap();

        // hide status bar
        window.set_decorations(false).unwrap();

        // focus the window
        window.set_focus().unwrap();

        // bring the window to front
        window.show().unwrap();

        // send focus to the window
        app_handle.emit_all(EVENT_SLIDE_FOCUS_INPUT, 1).unwrap();
    }
}

pub fn set_not_on_top(app_handle: &AppHandle, event: Event) {
    let window = app_handle.get_window("main").unwrap();
    window.set_always_on_top(false).unwrap();
    window.set_decorations(true).unwrap();
}